app/Factories/DashboardTileFactory.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  Created by Backbase on 12/03/2024.
+//
+
+import UIKit
+import DashboardJourneyCore
+import DashboardJourney
+import Resolver
+import RetailFeatureFilterUseCase
+import RetailAccountsAndTransactionsJourney
+
+class DashboardTileFactory {
+    // MARK: - Properties
+    @LazyInjected
+    private static var useCase: FeatureFilterUseCase
+    private static var allowedFeatures: [DashboardCore.Entitlement] = []
+    private static let featureMap: [String: UserEntitlement] = {
+        let values: [(String, UserEntitlement)] = (
+            QuickActionsTile.entitlements +
+            LatestTransactionsTile.entitlements
+        )
+            .map { $0.toUserEntitlement() }
+            .map { $0.toFeature() }
+
+        return Dictionary(uniqueKeysWithValues: values)
+    }()
+
+    private static func handleAllowedFeatures(_ features: [String]) {
+        Self.allowedFeatures = Self.featureMap
+            .filter { features.contains($0.key) }
+            .map { $0.value.toDashboardEntitlement() }
+    }
+
+    static func loadFeatures() async {
+        await withCheckedContinuation { continuation in
+            useCase.filterToCachedAllowedFeatures(Self.featureMap) { result in
+                switch result {
+                case let .success(allowedFeatures):
+                    Self.handleAllowedFeatures(allowedFeatures)
+                case .failure:
+                    Self.handleAllowedFeatures(Self.featureMap.map { $0.key })
+                }
+                continuation.resume()
+            }
+        }
+    }
+
+    private func checkEntitlements(_ entitlements: [DashboardCore.Entitlement]) -> Bool {
+        guard !entitlements.isEmpty else { return true }
+        return !Self.allowedFeatures.filter { entitlements.contains($0) }.isEmpty
+    }
+
+    // MARK: - Methods
+    func makeLatestTransactionsTile(_ parentNavigationController: UINavigationController) -> (any TileProtocol)? {
```

Comment:
These factories can be static as well to remove the need of creating an instance of this factory.